### ISSUE TITLE
Dataset issues

### ISSUE THREAD

**Issue by wj7486** [ali-vilab/ACE_plus#137]
I would like to confirm the construction of the training data, such as TryOn. According to the training instructions, six fields are required:

"edit_image": represents the input image for the editing task. If it is not an editing task but a reference generation, this field can be left empty.
"edit_mask": represents the input image mask for the editing task, used to specify the editing area. If it is not an editing task but rather for reference generation, this field can be left empty.
"ref_image": represents the input image for the reference image generation task;  if it is a pure editing task, this field can be left empty.
"target_image": represents the generated target image and cannot be empty.
"prompt": represents the prompt for the generation task.
"data_type": represents the type of data, which can be 'portrait', 'subject', or 'local'. This field is not used in training phase.

The six fields corresponding to the dataset I have prepared are as follows:
"edit_image": 
![Image](https://github.com/user-attachments/assets/843080ff-d588-4535-9f82-bd373d4e4b0e)
"edit_mask": 
![Image](https://github.com/user-attachments/assets/0a8762ab-27f3-41c4-ba16-ea460f7730b8)
"ref_image":
![Image](https://github.com/user-attachments/assets/ac681409-7650-47a5-91f9-d5545d8f5c2a)
"target_image":
![Image](https://github.com/user-attachments/assets/8aec3257-ce95-4069-a9d3-ace3430a3d09)
"prompt": 
Change the cloth in {image} to the one in {image1}
"data_type": 
subject

May I ask if the construction of this dataset is correct

**Comment by wj7486** [comment]
When training for 200 steps, the loss increases and the training fails to obtain the image:
edit_image0：
![Image](https://github.com/user-attachments/assets/ef11dd16-e455-49ab-a880-78b2250c395b)
modify_image0：
![Image](https://github.com/user-attachments/assets/bfd8af5b-b1e2-4b73-990a-3d3a2761185f)
edit_mask0
![Image](https://github.com/user-attachments/assets/fa57b18b-a3f1-43f2-ae20-2d09d0f1c3e3)
target_image
![Image](https://github.com/user-attachments/assets/7c701a40-0452-4860-84c5-f330985b3d52)
result:
![Image](https://github.com/user-attachments/assets/4dde8cdf-403f-4326-9ff3-0df579cb3290)

What problem is causing this? We look forward to your reply

**Comment by hsjkdjj** [comment]
> I would like to confirm the construction of the training data, such as TryOn. According to the training instructions, six fields are required:
> 
> "edit_image": represents the input image for the editing task. If it is not an editing task but a reference generation, this field can be left empty. "edit_mask": represents the input image mask for the editing task, used to specify the editing area. If it is not an editing task but rather for reference generation, this field can be left empty. "ref_image": represents the input image for the reference image generation task; if it is a pure editing task, this field can be left empty. "target_image": represents the generated target image and cannot be empty. "prompt": represents the prompt for the generation task. "data_type": represents the type of data, which can be 'portrait', 'subject', or 'local'. This field is not used in training phase.
> 
> The six fields corresponding to the dataset I have prepared are as follows: "edit_image": ![Image](https://github.com/user-attachments/assets/843080ff-d588-4535-9f82-bd373d4e4b0e) "edit_mask": ![Image](https://github.com/user-attachments/assets/0a8762ab-27f3-41c4-ba16-ea460f7730b8) "ref_image": ![Image](https://github.com/user-attachments/assets/ac681409-7650-47a5-91f9-d5545d8f5c2a) "target_image": ![Image](https://github.com/user-attachments/assets/8aec3257-ce95-4069-a9d3-ace3430a3d09) "prompt": Change the cloth in {image} to the one in {image1} "data_type": subject
> 
> May I ask if the construction of this dataset is correct

have you solved yet？

**Comment by wj7486** [comment]
> > 我想确认一下训练数据的构建，例如TryOn。根据训练说明，需要填写六个字段：
> > edit_image：表示编辑任务的输入图像。如果不是编辑任务而是参考生成，此字段可以留空。edit_mask：表示编辑任务的输入图像掩码，用于指定编辑区域。如果不是编辑任务而是参考生成，此字段可以留空。ref_image：表示参考图生成任务的输入图像；如果是纯编辑任务，此字段可以留空。target_image：表示生成的目标图像，不能为空。prompt：表示生成任务的提示信息。data_type：表示数据类型，可以是portrait、subject、local。此字段在训练阶段不使用。
> > 我准备好的数据集对应的六个字段如下： "edit_image": ![Image](https://github.com/user-attachments/assets/843080ff-d588-4535-9f82-bd373d4e4b0e)"edit_mask": ![图像](https://private-user-images.githubusercontent.com/117065644/456377060-0a8762ab-27f3-41c4-ba16-ea460f7730b8.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTMxNzkwMTAsIm5iZiI6MTc1MzE3ODcxMCwicGF0aCI6Ii8xMTcwNjU2NDQvNDU2Mzc3MDYwLTBhODc2MmFiLTI3ZjMtNDFjNC1iYTE2LWVhNDYwZjc3MzBiOC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwNzIyJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDcyMlQxMDA1MTBaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT05Mzc0MzBiZDc3YWRjYzA1ZDg2NjAyOTVlNDViNTY5Yzc2NDhjZTg3NTljY2FmOWIwMGU5OTNkMTdhN2JiNjI4JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.f3qkg8tRySv1aZJsz5CzUv6MMx15AdsZiCfAOPPHzRE)"ref_image": ![图像](https://private-user-images.githubusercontent.com/117065644/456377142-ac681409-7650-47a5-91f9-d5545d8f5c2a.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTMxNzkwMTAsIm5iZiI6MTc1MzE3ODcxMCwicGF0aCI6Ii8xMTcwNjU2NDQvNDU2Mzc3MTQyLWFjNjgxNDA5LTc2NTAtNDdhNS05MWY5LWQ1NTQ1ZDhmNWMyYS5qcGc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwNzIyJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDcyMlQxMDA1MTBaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1iNDk5ZGRlNTI5YTVkYWVhZGJiODI4ZTg4NzJiMzk4MWIxODQyZTAyMjUxNzkyZWUwN2Q1NjA1YmIyNWUwNzI4JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.ZU-YoB62p_7cSueHo_pRRYd79nhkspeFCxbx-1baRJU)"target_image": ![图像](https://private-user-images.githubusercontent.com/117065644/456377208-8aec3257-ce95-4069-a9d3-ace3430a3d09.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTMxNzkwMTAsIm5iZiI6MTc1MzE3ODcxMCwicGF0aCI6Ii8xMTcwNjU2NDQvNDU2Mzc3MjA4LThhZWMzMjU3LWNlOTUtNDA2OS1hOWQzLWFjZTM0MzBhM2QwOS5qcGc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwNzIyJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDcyMlQxMDA1MTBaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1mNWNiMWY5OGJkYjU1MTdkZTUwMjM3NjBjYzFjZjI0M2UyYjkxNWU5MzE4NTQ3NWE0NGJlOWIyNzVjM2Q2ZDk0JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.5lWMr5MGnarWFKd752Zsyfq6xTeRp9vf6ieNFPaZ18w)"prompt": 将{image}中的衣服换成{image1}中的衣服 "data_type": subject
> > 请问这个数据集的构造是否正确
> 
> 你解决了吗？


> > I would like to confirm the construction of the training data, such as TryOn. According to the training instructions, six fields are required:
> > "edit_image": represents the input image for the editing task. If it is not an editing task but a reference generation, this field can be left empty. "edit_mask": represents the input image mask for the editing task, used to specify the editing area. If it is not an editing task but rather for reference generation, this field can be left empty. "ref_image": represents the input image for the reference image generation task; if it is a pure editing task, this field can be left empty. "target_image": represents the generated target image and cannot be empty. "prompt": represents the prompt for the generation task. "data_type": represents the type of data, which can be 'portrait', 'subject', or 'local'. This field is not used in training phase.
> > The six fields corresponding to the dataset I have prepared are as follows: "edit_image": ![Image](https://github.com/user-attachments/assets/843080ff-d588-4535-9f82-bd373d4e4b0e) "edit_mask": ![Image](https://private-user-images.githubusercontent.com/117065644/456377060-0a8762ab-27f3-41c4-ba16-ea460f7730b8.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTMxNzkxNDAsIm5iZiI6MTc1MzE3ODg0MCwicGF0aCI6Ii8xMTcwNjU2NDQvNDU2Mzc3MDYwLTBhODc2MmFiLTI3ZjMtNDFjNC1iYTE2LWVhNDYwZjc3MzBiOC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwNzIyJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDcyMlQxMDA3MjBaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1jMjFjY2I0YTZkYWMzOWU1M2YwODczYWE0ODMzZjg0Yjg2ZjRkYzRkMDkwNmVhYmNjZmVlN2RkNjNmNTg1OGNmJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.Rq1q3QvpvCB5uKf0dbJhEanLysW1aWINZRPOcnZqXhY) "ref_image": ![Image](https://private-user-images.githubusercontent.com/117065644/456377142-ac681409-7650-47a5-91f9-d5545d8f5c2a.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTMxNzkxNDAsIm5iZiI6MTc1MzE3ODg0MCwicGF0aCI6Ii8xMTcwNjU2NDQvNDU2Mzc3MTQyLWFjNjgxNDA5LTc2NTAtNDdhNS05MWY5LWQ1NTQ1ZDhmNWMyYS5qcGc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwNzIyJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDcyMlQxMDA3MjBaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1mZjczMDAwYzIyNWEyOGM2MDZhYjRmODE1MmEzNGI3OGI2YTFlMmY0MjJkNzBiMzUwZjAxNTc1NGU1ZWZjMDJmJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.EOeS8heEGombsHprFbsfR7fvmnidNGefgv3-N2E5IZw) "target_image": ![Image](https://private-user-images.githubusercontent.com/117065644/456377208-8aec3257-ce95-4069-a9d3-ace3430a3d09.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTMxNzkxNDAsIm5iZiI6MTc1MzE3ODg0MCwicGF0aCI6Ii8xMTcwNjU2NDQvNDU2Mzc3MjA4LThhZWMzMjU3LWNlOTUtNDA2OS1hOWQzLWFjZTM0MzBhM2QwOS5qcGc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwNzIyJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDcyMlQxMDA3MjBaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0wYTFlZThiMmU0NDY4Mjc0MjBhZmNhM2RiOTMzYTdmZWU3MGMxYzlkZmJlMjQyNTI3ZjA0ZjU0ZWI2N2I2ZTQyJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.Nsv3PmpMJdhNkSktLtWrNhAe_3Yo3Ng1HGX33p9gPsk) "prompt": Change the cloth in {image} to the one in {image1} "data_type": subject
> > May I ask if the construction of this dataset is correct
> 
> have you solved yet？

解决了，通过增大CONTEXT_IN_DIM=4096

**Comment by wj7486** [comment]
如果你使用了ace++的subject lora, 确认一下配置文件中的参数是否正确添加，如：
CACHE_PRETRAIN_MODEL: True
COMFYUI_LORA_MODEL:  "comfyui_subject_lora16.safetensors"
